sysdeps/linux/generic/thread.cpp

```diff
@@ -31,20 +27,34 @@ extern "C" void __mlibc_enter_thread(void *entry, void *user_arg) {
 
 namespace mlibc {
 
-void *prepare_stack(void *entry, void *user_arg) {
-	uintptr_t map = reinterpret_cast<uintptr_t>(
-			mmap(nullptr, STACK_SIZE + GUARD_SIZE,
-				PROT_NONE,
-				MAP_PRIVATE | MAP_ANONYMOUS, -1, 0)
-			);
-	__ensure(reinterpret_cast<void*>(map) != MAP_FAILED);
-	int ret = mprotect(reinterpret_cast<void*>(map + GUARD_SIZE), STACK_SIZE,
-			PROT_READ | PROT_WRITE);
-	__ensure(!ret);
-
-	auto sp = reinterpret_cast<uintptr_t*>(map + STACK_SIZE + GUARD_SIZE);
+static constexpr size_t default_stacksize = 0x200000;
+
+int sys_prepare_stack(void **stack, void *entry, void *user_arg, void *tcb, size_t *stack_size, size_t *guard_size) {
+	(void)tcb;
+	if (!*stack_size)
+		*stack_size = default_stacksize;
```

Comment:
It'd be nice to use camelCasd in these functions, but maybe we should leave it to a future PR to fix that.